testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetITSupport.java

```diff
@@ -109,7 +112,7 @@ protected NugetClient nugetClient(final Repository repository) throws Exception
   }
 
   @Nonnull
-  private String prepareApiKey(final PrincipalCollection admin) {
+  private synchronized String prepareApiKey(final PrincipalCollection admin) {
```

Comment:
Won't `synchronized` here potentially skew the results or mask concurrency issues in the code under test? One option might be to store the current apiKey in a `volatile` which is checked outside of the `synchronized` then checked again inside (in case another thread got there first) and only then call `createApiKey` and cache the result.